app/services/identity_linker.rb

```diff
@@ -18,12 +20,15 @@ def link_identity(
     clear_deleted_at: nil
   )
     return unless user && service_provider.present?
+
     process_ial(ial)
+    process_aal(aal)
```

Comment:
I think this shouldn't be a separate method, `process_ial` has some other weird side effecrts for other columns, but `@aal` is just a straightfoward ivar set, I think it would be clearer to set it directly